book.md

```diff
@@ -792,15 +792,23 @@ forward-edge CFI\index{forward-edge CFI}, whereas a scheme that protects
 returns is said to implement backward-edge CFI\index{backward-edge CFI}.
 
 Ideally, a CFI scheme would not allow any control flow transfers that don't
-occur in a correct program execution, however different schemes have varying
-granularities.
-
-CFI schemes are sometimes classified as coarse-grained\index{coarse-grained CFI}
-or fine-grained\index{fine-grained CFI}. While there is no precise definition of
-these terms, coarse-grained CFI schemes typically have relatively coarse
-granularity. For example, a CFI scheme that allows an indirect function call to
-continue the execution at the start of any function, rather than only at the
-start of functions with a specific signature, is considered coarse-grained.
+occur in a correct program execution. However, different schemes have varying
+granularities. In general, the legal branch targets will be divided into
+classes, with targets in each class treated as equivalent for security purpose.
```

Comment:
We privately talked about how the concept of "equivalent classes" is introduced in the very next section.
This paragraph now also explains dividing legal branch targets into classes.
I think this slight repetition is totally fine though in this case.
It's sometimes works well to explain a concept twice in slightly different words when the text is meant to educate. I think in this case it works well.